pkg/logs/tekton_logging.go

```diff
@@ -218,6 +218,10 @@ func GetRunningBuildLogs(pa *v1.PipelineActivity, buildName string, kubeClient k
 		if !foundLogs {
 			break
 		}
+		pipelineRunNames, err = getPipelineRunNamesForActivity(pa, tektonClient)
```

Comment:
So when the meta pipeline starts we get the first set of pipeline runs which are getting streamed and once they are all streamed before we exit the loop we get the names again which will capture the pipeline runs for execution pipeline. This works since the last step of the meta pipeline is the creation of the execution pipeline CRDs. Right?

It is still all very implicit not based on non explicit assumptions. I also wonder whether there is not a potential race condition. I wonder whether we should bake the notion of a meta pipeline better into the code. One thing we could for example to is add a label which differentiates between pipeline run for meta vs execution pipeline. This way, given a pipeline activity one can explicity ask first for the meta pipeline pipeline runs and then for the execution pipeline runs. WDYT? Worth opening an issue? @dgozalo?

Reply:
See #5007.